lib/ssl/src/ssl_handshake.erl

```diff
@@ -596,7 +599,7 @@ encode_handshake(#certificate_verify{signature = BinSig, hashsign_algorithm = Ha
 encode_handshake(#finished{verify_data = VerifyData}, _Version) ->
     {?FINISHED, VerifyData}.
 
-encode_hello_extensions(_, {3, 0}) ->
+encode_hello_extensions(_, ?'SSL-3.0') ->
```

Comment:
This clause should no longer be needed! I would remove only this clause and then run the test cases.